docs/references/cmcspec.mdx

```diff
@@ -0,0 +1,241 @@
+- Depositing minted cycles into ledger-managed accounts
+- Fetching exchange rates and subnet configuration
+
+The CMC is governed entirely by the **NNS (Network Nervous System)** and receives configuration updates through proposals.
+
+## Configuration Parameters
+
+- **Conversion Rate Source**: The ICP/XDR exchange rate is **pushed** into the CMC by a dedicated *exchange rate canister*. This data reflects live market pricing and determines the ICP→cycles conversion.
+
+- **Governance Control**: All operations of the CMC — including upgrades, configuration, and authorized callers — are managed through NNS proposals.
+
+
+## API Endpoints
+
+The CMC exposes a set of core methods for converting ICP into cycles and interacting with subnet configuration. These include:
+
+- `notify_create_canister`: Processes an ICP payment by minting cycles and using them to create a new canister, assigning control to the specified principal and applying optional settings.
+- `notify_top_up`: Processes an ICP payment by minting cycles and sending them to an existing canister to increase its available balance.
+- `notify_mint_cycles`: Processes an ICP payment by minting cycles and depositing them into a cycles ledger account associated with a subaccount.
+- `create_canister`: Creates a canister using cycles directly attached to the call.
+- `get_icp_xdr_conversion_rate`: Returns the current ICP/XDR exchange rate with certification.
+- `get_subnet_types_to_subnets`: Lists available subnets grouped by their types.
+- `get_principals_authorized_to_create_canisters_to_subnets`: Indicates which principals are permitted to create canisters on which subnets.
+- `get_default_subnets`: Returns the subnets for general-purpose canister creation.
+- `get_build_metadata`: Displays internal version and build information for the CMC.
+
+
+## API Reference
+
+### `notify_create_canister`
+
+Creates a new canister after verifying an ICP payment recorded in the ICP Ledger.
+If the `subnet_selection field is omitted, the CMC will automatically choose a suitable subnet at random from the available options.
+
+The CMC expects the payment to follow a strict structure that encodes both the **intent** and the **recipient**:
+
+- The **destination account** of the ICP transfer must be the CMC's account with a subaccount derived from the intended controller's principal.
+- The **memo field** must explicitly indicate the intent to create a canister. This can be expressed as:
+  - A legacy 64-bit unsigned integer memo with value `0x41455243` (ASCII `"CREA"`)
+  - Or, for ICRC-1-compatible transfers (e.g., `icrc1_transfer`, `icrc2_transfer_from`), a memo blob equal to:
+    ```candid
+    "\43\52\45\41\00\00\00\00"
+    ```
+    which represents `"CREA"` padded to 8 bytes.
+
+- **Parameters:**
+  ```candid
+  record {
+    block_index: nat64;              // The ledger block containing the ICP payment
+    controller: principal;           // The principal who will control the new canister
```

Comment:
```suggestion
    controller: principal;           // The creator of the canister. Must match caller; otherwise, Err is returned. This is also used when checking that the creator is authorized to create canisters in subnets where authorization is required. This is also used when `settings` does not specify `controllers`.
```